daemon/delete.go

```diff
@@ -47,15 +47,16 @@ func (daemon *Daemon) ContainerRm(name string, config *ContainerRmConfig) error
 				logrus.Debugf("Could not update network to remove link %s: %v", n, err)
 			}
 		}
-
-		return nil
```

Comment:
not sure about this, `docker rm --link` is about just removing a link between two containers and this seems incorrect, I hope there's some test covering this also